### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by ShaliniNayan**
Hello,

For this PR, I have an OOP school library and add unit tests to the the codebase, specifically to classes for the project using the following:

- Add unit tests for all the methods in all the classes:
   - Book
   - Person
   - Student
   - Teacher
   - Classroom
   - Rental
   - all classes that were created while implementing the Decorator design pattern.

- No linters error.

- Gitflow follows.

### CODE REVIEW

File: book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    it 'should have the correct title and author' do
+      expect(book.title).to eq('Title')
+      expect(book.author).to eq('Author')
+    end
+
+    it 'should have an empty rentals array' do
+      expect(book.rentals).to be_empty
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline book_spec.rb]: - [x] kindly note that you are to test methods in a class too, so please add  a test case for method `add_rental`

File: person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  let(:person) { Person.new(parent_permission: true, name: 'John', age: 25) }
+
+  context 'when initializing a person' do
+    it 'should have the correct name, age, and parent_permission' do
+      expect(person.parent_permission).to be(true)
+      expect(person.name).to eq('John')
+      expect(person.age).to eq(25)
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline person_spec.rb]: - [x] kindly add tests for methods like `correct_name`, `can_use_services?`, `add_rental`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @ShaliniNayan  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**